src/Common/Core/Impl/Services/IServiceManager.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+
+namespace Microsoft.Common.Core.Services {
+    public interface IServiceManager: IServiceContainer, IDisposable {
+        /// <summary>
+        /// Adds service instance
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="service">Service instance</param>
+        IServiceContainer AddService<T>(T service) where T : class;
+
+        /// <summary>
+        /// Adds on-demand created service
+        /// </summary>
+        /// <param name="type">Service type</param>
+        /// <param name="factory">Optional service factory. If not provided, reflection with default constructor will be used.</param>
+        /// <param name="factoryParameters">Optional arguments to the factory call.</param>
+        IServiceContainer AddService(Type type, IServiceFactory factory = null, params object[] factoryParameters);
```

Comment:
In this case. it should return `IServiceManager` rather than `IServiceContainer`.

---
In reply to: [106053919](https://github.com/Microsoft/RTVS/pull/3302#discussion_r106053919) [](ancestors = 106053919)